feat: enable bypassing persisted operations (#4)
graphile/persisted-operations: README.md

This plugin adds the following options to the PostGraphile library options:

> ````ts

Reviewer: This was likely Prettier escaping a code block inside a code block; I don't think you need it so please go back to just 3 backticks
````suggestion
```ts
````

 * stores (e.g. S3).
 */
persistedOperationsGetter?: PersistedOperationGetter;

/**

>  * In development it's common to want to send arbitrary queries from GraphiQL
>  * whilst also enforcing Persisted Operations from the application. You should
>  * use this function if you want unpersisted operation to be allowed in some
>  * context (e.g in development).

Reviewer: ```suggestion
 * There are situations where you may want to allow arbitrary operations
 * (for example using GraphiQL in development, or allowing an admin to
 * make arbitrary requests in production) whilst enforcing Persisted
 * Operations for the application and non-admin users. This function
 * allows you to determine under which circumstances persisted operations
 * may be bypassed.
```

>  * @example
>  * app.use(postgraphile(DATABASE_URL, SCHEMAS, {
>  *   allowUnpersistedOperation(req) {
>  *     return process.env.NODE_ENV === "development" && req.headers.referer.endsWith("/graphiql");
>  *   }
>  * });

Reviewer: ```suggestion
 *
 * @example
 *
 * ```
 * app.use(postgraphile(DATABASE_URL, SCHEMAS, {
 *   allowUnpersistedOperation(req) {
 *     return process.env.NODE_ENV === "development" && req.headers.referer?.endsWith("/graphiql");
 *   }
 * });
 * ```
```

 */

> allowUnpersistedOperation?(request: any): boolean;

Reviewer: I think the type for this is something like `IncomingMessage`? You'll also want the payload here:

```suggestion
allowUnpersistedOperation?(request: IncomingMessage, payload: any}): boolean;
```

 * stores (e.g. S3).
 */
persistedOperationsGetter?: PersistedOperationGetter;

/**
 * There are situations where you may want to allow arbitrary operations
 * (for example using GraphiQL in development, or allowing an admin to
 * make arbitrary requests in production) whilst enforcing Persisted
 * Operations for the application and non-admin users. This function
 * allows you to determine under which circumstances persisted operations

>  * may be bypassed.

Reviewer: ```suggestion
 * may be bypassed.
 *
 * IMPORTANT: this function must not throw!
```

 *
 * @example
 *
 * ```
 * app.use(postgraphile(DATABASE_URL, SCHEMAS, {
 *   allowUnpersistedOperation(req) {
 *     return process.env.NODE_ENV === "development" && req.headers.referer?.endsWith("/graphiql");
 *   }
 * });
 * ```
 */ 
 allowUnpersistedOperation?: boolean | ((request: IncomingMessage, payload: any): boolean);

> ````

Reviewer: ````suggestion
```
````
